features/fields.feature

```diff
@@ -0,0 +1,341 @@
+      | burgerservicenummer | 999995078                                                       |
+      | fields              | burgerservicenummer,geslachtsaanduiding.code,naam.geslachtsnaam |
+      Dan heeft de persoon met burgerservicenummer '999995078' alleen de volgende gegevens
+      | naam                     | waarde    |
+      | burgerservicenummer      | 999995078 |
+      | geslachtsaanduiding.code | M         |
+      En heeft de persoon met burgerservicenummer '999995078' alleen de volgende 'naam' gegevens
+      | naam          | waarde  |
+      | geslachtsnaam | Groenen |
+
+    Scenario: Gevraagd veld is een gegeven van een gegevensgroep collectie van de persoon
+      Als personen wordt gezocht met de volgende parameters
+      | naam                | waarde                                                         |
+      | type                | RaadpleegMetBurgerservicenummer                                |
+      | burgerservicenummer | 999995078                                                      |
+      | fields              | burgerservicenummer,partners.type,partners.burgerservicenummer |
+      Dan heeft de persoon met burgerservicenummer '999995078' alleen de volgende gegevens
+      | naam                | waarde    |
+      | burgerservicenummer | 999995078 |
+      En heeft de persoon met burgerservicenummer '999995078' een 'partner' met alleen de volgende gegevens
+      | naam                | waarde    |
+      | type                | Partner   |
+      | burgerservicenummer | 999992935 |
+
+    Abstract Scenario: Gevraagd veld heeft geen waarde voor de persoon
+      Als personen wordt gezocht met de volgende parameters
+      | naam                | waarde                          |
+      | type                | RaadpleegMetBurgerservicenummer |
+      | burgerservicenummer | 999995078                       |
+      | fields              | burgerservicenummer,<veld>      |
+      Dan heeft de persoon met burgerservicenummer '999995078' alleen de volgende gegevens
+      | naam                | waarde    |
+      | burgerservicenummer | 999995078 |
+
+      Voorbeelden:
+      | veld                            |
+      | geheimhoudingPersoonsgegevens   |
+      | naam.adellijkeTitelPredicaat    |
+      | inOnderzoek.geslachtsaanduiding |
+
+  Rule: optioneel mag je (het begin van) het pad weglaten wanneer dit uniek verwijst naar 1 veld in de resource
+    - wanneer het opgegeven pad exact verwijst naar een veld en tegelijkertijd als deel van een pad verwijst naar een ander veld, wordt alleen het veld met het exacte pad opgenomen in de response
+    - wanneer het opgegeven pad niet exact verwijst naar een veld en tegelijkertijd als deel van een pad verwijst naar meerdere ander velden, wordt een foutmelding gegeven
+    - wanneer het opgegeven pad verwijst naar een veld en tegelijkertijd naar het gelijknamige veld in inOnderzoek, worden beide velden opgenomen in de response wanneer ze een waarde hebben

```

Comment:
Ondanks dat ik me voor kan stellen dat je dat juist niet wil toch maar even de vraag of je deze situatie niet ook terug wil zien in het csv bestand?

In dat geval zou je dus in het record waarvan de eerste kolom de waarde 'indicatieCurateleRegister' heeft de tweede kolom de waarde 'gezagsverhouding.indicatieCurateleRegister,gezagsverhouding.inOnderzoek.indicatieCurateleRegister' moeten geven.

Reply:
> Klopt het dat ik de csv zo moet lezen dat als de field parameter een waarde bevat die gelijk is aan een waarde in de eerste kolom dan wordt deze geïnterpreteerd als had hij de waarde in de tweede kolom?
> 
> Waarmee de waarde 'indicatieCurateleRegister' nooit geïnterpreteerd kan worden als 'gezagsverhouding.inOnderzoek.indicatieCurateleRegister'.

Ik zie dat het antwoord op deze vraag al in het fields.feature bestand staat.

> Klopt het ook dat meerdere waarden in de eerste kolom naar eenzelfde waarde in de tweede kolom kunnen leiden? Dit lijkt me het geval bij 'aanschrijfwijze.aanspreekvorm' en 'aanspreekvorm', beiden kunnen geïnterpreteerd worden als 'naam.aanschrijfwijze.aanspreekvorm'.

Blijft alleen nog deze vraag open staan.

Reply:
ik wilde de tabel nog enigszins eenvoudig en leesbaar houden. het doel van de csv tabel is een overzicht van de vertaling van gedeeltelijk pad naar volledig pad. Ik denk dat dat vertroebelt wanneer er ook andere regels in verwerkt worden.